RecoCTPPS/TotemRPLocal/src/TotemTimingRecHitProducerAlgorithm.cc

```diff
@@ -10,65 +10,69 @@
 #include "RecoCTPPS/TotemRPLocal/interface/TotemTimingRecHitProducerAlgorithm.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
+#include <numeric>
+
 //----------------------------------------------------------------------------------------------------
 
-const float TotemTimingRecHitProducerAlgorithm::SINC_COEFFICIENT = std::acos(-1) * 2 / 7.8;
+const float TotemTimingRecHitProducerAlgorithm::SINC_COEFFICIENT = M_PI*2 / 7.8;
 
 //----------------------------------------------------------------------------------------------------
 
-TotemTimingRecHitProducerAlgorithm::TotemTimingRecHitProducerAlgorithm(
-    const edm::ParameterSet &iConfig)
-    : sampicConversions_(iConfig.getParameter<std::string>("calibrationFile")),
-      baselinePoints_(iConfig.getParameter<int>("baselinePoints")),
-      saturationLimit_(iConfig.getParameter<double>("saturationLimit")),
-      cfdFraction_(iConfig.getParameter<double>("cfdFraction")),
-      smoothingPoints_(iConfig.getParameter<int>("smoothingPoints")),
-      lowPassFrequency_(iConfig.getParameter<double>("lowPassFrequency")),
-      hysteresis_(iConfig.getParameter<double>("hysteresis")) {}
-
-void TotemTimingRecHitProducerAlgorithm::build(
-    const CTPPSGeometry *geom, const edm::DetSetVector<TotemTimingDigi> &input,
-    edm::DetSetVector<TotemTimingRecHit> &output) {
-  for (const auto &vec : input) {
+TotemTimingRecHitProducerAlgorithm::TotemTimingRecHitProducerAlgorithm(const edm::ParameterSet& iConfig) :
+  sampicConversions_(iConfig.getParameter<bool>("mergeTimePeaks"),
+                     iConfig.getParameter<edm::FileInPath>("calibrationFile").fullPath()),
+  baselinePoints_  (iConfig.getParameter<int>   ("baselinePoints")),
+  saturationLimit_ (iConfig.getParameter<double>("saturationLimit")),
+  cfdFraction_     (iConfig.getParameter<double>("cfdFraction")),
+  smoothingPoints_ (iConfig.getParameter<int>   ("smoothingPoints")),
+  lowPassFrequency_(iConfig.getParameter<double>("lowPassFrequency")),
+  hysteresis_      (iConfig.getParameter<double>("hysteresis"))
+{}
+
+void
+TotemTimingRecHitProducerAlgorithm::build(const CTPPSGeometry* geom, const edm::DetSetVector<TotemTimingDigi>& input,
+                                          edm::DetSetVector<TotemTimingRecHit>& output)
+{
+  for (const auto& vec : input) {
     const TotemTimingDetId detid(vec.detId());
 
-    float x_pos = 0.0f, y_pos = 0.0f, z_pos = 0.0f, x_width = 0.0f, y_width = 0.0f, 
-          z_width = 0.0f;
+    float x_pos = 0., y_pos = 0., z_pos = 0.;
+    float x_width = 0., y_width = 0., z_width = 0.;
```

Comment:
why were the rhs literals changed from floats (matching the lhs type) to doubles?

Reply:
Indeed, my mistake! Reverted...